apps/webapp/shared/input-token/select-token-modal.tsx

```diff
@@ -0,0 +1,72 @@
+'use client';
+
+import React, { Dispatch, SetStateAction, useMemo, useState } from 'react';
+import { Dialog, DialogContent, DialogHeader, DialogPrimitive, DialogTrigger, Input } from 'ui';
+import { ResponsiveImage } from '../responsive-image';
+import { Asset } from '../../app/send/types';
+import { MagnifyingGlassIcon } from '@radix-ui/react-icons';
+import { assets } from '../../app/send/constants';
+
+interface SelectTokenModalProps {
+  asset: Asset;
+  setAsset: Dispatch<SetStateAction<Asset>>;
+}
+
+export const SelectTokenModal = ({ asset, setAsset }: SelectTokenModalProps) => {
+  const [search, setSearch] = useState('');
+
+  const filteredAsset = useMemo(() => {
+    if (!search) return assets;
+    return assets.filter(asset => asset.name.toLowerCase().includes(search.toLowerCase()));
+  }, [search]);
```

Comment:
Can you also sort this to where the assets with the most funds are at the top? And then if zero in the wallet, it can be alphabetical.